autorest/azure/devicetoken.go

```diff
@@ -86,12 +89,14 @@ type deviceToken struct {
 
 // InitiateDeviceAuth initiates a device auth flow. It returns a DeviceCode
 // that can be used with CheckForUserCompletion or WaitForUserCompletion.
-func InitiateDeviceAuth(client *autorest.Client, clientID, resource string) (*DeviceCode, error) {
+func InitiateDeviceAuth(client *autorest.Client, clientID, tenantID, resource string) (*DeviceCode, error) {
+	oAuthDeviceEndpoint := strings.Replace(OAuthDeviceEndpointTemplate, "{tenantId}", tenantID, -1)
```

Comment:
I think we should either make it `%s` or text/template. string.Replace seems very hacky.